src/sharedmemory_com.c

```diff
@@ -56,15 +56,14 @@ int smc_write(uint8_t data)
   {
     QUARK_BUFF[QUARK_BUFF_HEAD] = data;
     QUARK_BUFF_HEAD = new_head;
+    QUARK_BUFF_FLAG = 2; //unlock the buffer
```

Comment:
It'll be a little nicer to just have one return statement, and one assignment of QUARK_BUFF_FLAG, i.e.
`...`
`QUARK_BUFF_FLAG = 2; // unlock the buffer`
`return ret;`
`}`